lib/sync/remote-resin-io-device.coffee

```diff
@@ -128,20 +131,39 @@ exports.sync = ({ uuid, baseDir, destination, before, after, ignore, port = 22,
 	throw new Error("'destination' is a required sync option") if not destination?
 	throw new Error("'uuid' is a required sync option") if not uuid?
 
+	# Resolves with object with required device info or is rejected if API was not accessible. Resolved object:
+	#
+	#	{
+	#		fullUuid: <string, full resin.io device UUID>
+	#		isAUFS: <true|false, depending on if the device is running on AUFS>
+	#	}
+	#
 	getDeviceInfo = (uuid) ->
+		RequiredDeviceObjectFields = [ 'uuid', 'os_version' ]
+
+		# Returns a promise that is resolved with the API-fetched device object or rejected on error or missing requirement
+		ensureDeviceRequirements = (device) ->
+			# Ensure user is the owner of the device. This is also checked on the backend side.
+			resin.auth.getUserId()
+			.then (userId) ->
+				if userId isnt device.user.__id
+					throw new Error('Resin sync is permitted to the device owner only. The device owner is the user who provisioned it.')
+			.then -> # Ensure minimum required HostOS version for resin sync
+				ensureHostOSCompatibility(device.os_version, MIN_HOSTOS_RSYNC)
+			.then ->
+				throw new Error("Fetched device info is missing required field #{field}") for field in RequiredDeviceObjectFields when field not of device
```

Comment:
This is a really long line, I'd at least like to see it broken up onto multiple lines, however you could also look at using lodash and reporting all the missing fields at once (rather than having to fix one at a time), eg
```coffee
missingKeys = _.difference(RequiredDeviceObjectFields, _.keys(device))
if missingKeys.length > 0
	throw new Error("Fetched device info is missing required fields '#{missingKeys.join("', '")}'")
```